docs/book/src/reference/attributes.md

```diff
@@ -102,3 +103,34 @@ The `#[test]` attribute marks a function to be executed as a test.
 The `#[test(should_revert)]` attribute marks a function to be executed as a test that should revert.
 
 More details in [Unit Testing](../testing/unit-testing.md).
+
+## ABI Name
+
+The `#[abi_name]` attribute allows to specify the ABI name for an item.
+This means that when a contract ABI JSON file is generated, the name that is output is the one specified
+by the attribute. This can be useful to allow renaming items, while allowing for keeping backwards
+compatibility at the contract ABI level.
```

Comment:
I think this needs a larger example. With at least concrete mention of how this changes the ABI output.